app/stack/log-drains/new/controller.js

```diff
@@ -0,0 +1,26 @@
+import Ember from "ember";
+
+// We create an anchor since that is, afaik, the easiest way to parse a url in javascript
+function parseUrl(url) {
+    let a = document.createElement('a');
+    a.href = url;
+    return a;
+}
+
+export default Ember.Controller.extend({
+    actions: {
+    setDrainFromDatabase(database) {
```

Comment:
I tried to move that handler into the route but then I got a whole bunch of errors. I guess that is due to it not having access to the model?